packages/cli/src/util/token.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { consola } from "consola";
+import * as fs from "node:fs";
+import path from "node:path";
+
+const TOKEN_ENV_VARS = ["FOUNDRY_TOKEN", "FOUNDRY_SDK_AUTH_TOKEN"] as const;
+
+/**
+ * Loads a JWT Auth Token from an argument, a file, or environment variable returning the first found.
+ * @param token The token as a string.
+ * @param tokenFile The path to the token file.
+ * @returns The token as a string.
+ * @throws An error if no token is found.
+ */
+export function loadToken(token?: string, tokenFile?: string): string {
+  if (token) {
+    validate(token);
+    consola.debug(`Using token from --token argument`);
+    return token;
+  }
+
+  if (tokenFile) {
+    const loadedToken = loadTokenFile(tokenFile);
+    validate(loadedToken);
+    consola.debug(`Using token from --tokenFile=${tokenFile} argument`);
```

Comment:
Same comment here about logging we're using the token from file as soon as we know it. Also we may want to log the full resolved path for debugging as I guess this may be a relative path in the input